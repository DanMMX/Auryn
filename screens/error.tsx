import * as React from 'react';
import { View, StyleSheet, Text } from 'react-native';

interface ErrorProps {
  text: string;
}

export const Error: React.FunctionComponent<ErrorProps> = ({ text }) => (
  <View style={styles.container}>
    <Text style={styles.text}>{text}</Text>
  </View>
);

const styles = StyleSheet.create({
  container: {
    flex: 1,
    borderColor: 'black',
    borderWidth: 3,
    backgroundColor: '#143672',
    alignItems: 'center',
    justifyContent: 'center',
    flexWrap: 'wrap',
  },
  text: {
    fontSize: 36,
    color: 'white',
  },
});
