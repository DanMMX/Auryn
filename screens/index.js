import Splash   from './splash.youi'
import Lander   from './lander.youi'
import PDP      from './pdp.youi'
import Player   from './video.youi'

export {
  Splash,
  Lander,
  PDP,
  Player,
}
