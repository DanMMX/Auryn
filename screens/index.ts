/**
 * Copyright (c) You i Labs Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import Splash from './splash';
import Lander from './lander';
import PDP from './pdp';
import Search from './search';
import Profile from './profile';
import Video from './video';

export {
  Splash,
  Lander,
  PDP,
  Search,
  Profile,
  Video,
};
