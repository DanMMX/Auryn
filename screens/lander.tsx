/**
 * Copyright (c) You i Labs Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import * as React from 'react';
import { Composition, ViewRef, ScrollRef, ButtonRef, FocusManager, BackHandler } from '@youi/react-native-youi';
import { View, NativeEventSubscription } from 'react-native';
import { Timeline, ToggleGroup, List } from '../components';
import {
  withNavigationFocus,
  NavigationActions,
  NavigationScreenProps,
  NavigationEventSubscription,
} from 'react-navigation';
import { connect, DispatchProp } from 'react-redux';
import { tmdb } from '../actions';
import { Asset } from '../adapters/asset';
import { Config } from '../config';
import { TmdbActionTypes } from '../typings/tmdbReduxTypes';
import { AurynAppState } from '../reducers';
import { ListItemFocusEvent, ListItemPressEvent } from '../components/listitem';
import { ToggleButtonPress } from '../components/toggleButton';

interface LanderProps extends NavigationScreenProps, DispatchProp<TmdbActionTypes> {
    isFocused: boolean;
    tv: Asset[];
    discover: Asset[];
    movies: Asset[];
  };

interface LanderState {
  currentListIndex: number;
}

class Lander extends React.Component<LanderProps, LanderState> {
  state = { currentListIndex: 0 };

  lists: React.RefObject<List>[] = Array(4).fill(React.createRef<List>());

  lastFocusItem = React.createRef<ButtonRef>();

  lastFocusNavItem = React.createRef<ButtonRef>();

  navButtonNames: string[] = ['Discover', 'Movies', 'Shows', 'Live'];

  focusListener!: NavigationEventSubscription;

  blurListener!: NavigationEventSubscription;

  backHandlerListener!: NativeEventSubscription;

  outTimeline = React.createRef<Timeline>();

  navInTimeline = React.createRef<Timeline>();

  inTimeline = React.createRef<Timeline>();

  menuButtons = React.createRef<ToggleGroup>();

  searchButton = React.createRef<ButtonRef>();

  profileButton = React.createRef<ButtonRef>();

  scroller = React.createRef<ScrollRef>();

  componentDidMount() {
    this.focusListener = this.props.navigation.addListener('didFocus', () => {
      this.backHandlerListener = BackHandler.addEventListener('hardwareBackPress', this.navigateBack);

      if (this.lastFocusNavItem && this.lastFocusNavItem.current) {
        FocusManager.enableFocus(this.lastFocusNavItem.current);
        FocusManager.focus(this.lastFocusNavItem.current);
      } else if (this.lastFocusItem && this.lastFocusItem.current) {
        FocusManager.enableFocus(this.lastFocusItem.current);
        FocusManager.focus(this.lastFocusItem.current);
      }

      if (this.inTimeline.current && this.navInTimeline.current) {
        this.navInTimeline.current.play();
        this.inTimeline.current.play();
      }
    });
    this.blurListener = this.props.navigation.addListener('didBlur', () => this.backHandlerListener.remove());

    if (this.menuButtons.current) FocusManager.focus(this.menuButtons.current.getButtonRef(0).current);
  }

  navigateBack = () => {
    if (this.menuButtons.current)
      FocusManager.focus(this.menuButtons.current.getButtonRef(this.state.currentListIndex).current);
    return true;
  };

  componentWillUnmount() {
    this.focusListener.remove();
    this.blurListener.remove();
    this.backHandlerListener.remove();
  }

  navigateToScreen = async (screen: string) => {
    const navigateAction = NavigationActions.navigate({
      routeName: screen,
    });

    if (screen === 'Search' && this.searchButton.current) this.lastFocusNavItem = this.searchButton;
    else if (screen === 'Profile' && this.profileButton.current) this.lastFocusNavItem = this.profileButton;

    if (this.outTimeline.current) await this.outTimeline.current.play();
    this.props.navigation.dispatch(navigateAction);
  };

  scrollToViewByIndex: ToggleButtonPress = index => {
    if (!Config.isRoku) {
      if (this.menuButtons.current) {
        for (let i = 0; i < this.lists.length; i++)
        FocusManager.setNextFocus(this.menuButtons.current.getButtonRef(i).current, this.lists[index].current, 'down');
      }

      if (this.menuButtons.current
        && this.searchButton.current
        && this.profileButton.current
        && this.lists[index].current
      ) {
        FocusManager.setNextFocus(this.searchButton.current, this.lists[index].current, 'down');
        FocusManager.setNextFocus(this.profileButton.current, this.lists[index].current, 'down');
        FocusManager.setNextFocus(this.lists[index].current, this.menuButtons.current.getButtonRef(index).current, 'up');
      }
    }

    this.setState({ currentListIndex: index });
    if (this.scroller.current) {
      this.scroller.current.scrollTo({
        x: 0,
        y: (index * 900) + 1, // eslint-disable-line no-extra-parens
        animated: true,
      });
    }
  };

  // eslint-disable-next-line max-params
  onFocusItem: ListItemFocusEvent = (id, type, ref, shouldChangeFocus) => {
    this.props.dispatch(tmdb.prefetchDetails(id, type));
    this.lastFocusItem = ref;

    if (shouldChangeFocus === false || Config.isRoku || !ref.current) return;

    if (this.menuButtons.current) {
      FocusManager.setNextFocus(ref.current, this.menuButtons.current.getButtonRef(this.state.currentListIndex).current, 'up');
      for (let index = 0; index < this.lists.length; index++)
        FocusManager.setNextFocus(this.menuButtons.current.getButtonRef(index).current, ref.current, 'down');
    }

    if (this.searchButton.current && this.profileButton.current) {
      FocusManager.setNextFocus(this.searchButton.current, ref.current, 'down');
      FocusManager.setNextFocus(this.profileButton.current, ref.current, 'down');
    }
  };

  onPressItem: ListItemPressEvent = async (id, type, ref) => {
    this.lastFocusItem = ref;
    this.lastFocusNavItem = null;
    const navigateAction = NavigationActions.navigate({
      routeName: 'PDP',
      params: {
        id,
        type,
      },
    });
    this.props.dispatch(tmdb.getDetailsByIdAndType(id, type));
    if (this.outTimeline.current) await this.outTimeline.current.play();
    this.props.navigation.dispatch(navigateAction);
  };

  // eslint-disable-next-line max-lines-per-function
  render() {
    const { isFocused, tv, movies, discover } = this.props;
    const { currentListIndex } = this.state;

    // null list is used for Roku
    const nullList = (
      <Composition source="Auryn_Container-NullList">
        <List name="NullList" type="None" data={[]} focusable={false} />
      </Composition>
    );

    const lists = [
      <Composition source="Auryn_Container-Discover" key="discover">
        <List
          name="Discover"
          type="Discover"
          data={discover}
          ref={this.lists[0]}
          focusable={isFocused && currentListIndex === 0}
          onFocusItem={this.onFocusItem}
          onPressItem={this.onPressItem}
        />
      </Composition>,
      <Composition source="Auryn_Container-Movies" key="movies">
        <List
          name="Movies"
          type="Movies"
          data={movies}
          ref={this.lists[1]}
          focusable={isFocused && currentListIndex === 1}
          onFocusItem={this.onFocusItem}
          onPressItem={this.onPressItem}
        />
      </Composition>,
      <Composition source="Auryn_Container-Shows" key="shows">
        <List
          name="Shows"
          type="Shows"
          data={tv}
          ref={this.lists[2]}
          focusable={isFocused && currentListIndex === 2}
          onFocusItem={this.onFocusItem}
          onPressItem={this.onPressItem}
        />
      </Composition>,
      <Composition source="Auryn_Container-Live" key="live">
        <List
          name="Live"
          type="Live"
          data={movies.slice(0, 2)}
          ref={this.lists[3]}
          focusable={isFocused && currentListIndex === 3}
          onFocusItem={this.onFocusItem}
          onPressItem={this.onPressItem}
        />
      </Composition>,
    ];
    return (
      <Composition source="Auryn_Lander">
        <ToggleGroup
          focusable={isFocused}
          prefix="Btn-Nav-"
          names={this.navButtonNames}
          onPressItem={this.scrollToViewByIndex}
          ref={this.menuButtons}
        />
        <ButtonRef
          name="Btn-Nav-Search"
          focusable={isFocused}
          ref={this.searchButton}
          onPress={() => this.navigateToScreen('Search')}
        />
        <ButtonRef
          name="Btn-Nav-Profile"
          focusable={isFocused}
          ref={this.profileButton}
          onPress={() => this.navigateToScreen('Profile')}
        />
        <Timeline name="LanderIn" ref={this.inTimeline} />
        <Timeline name="LanderOut" ref={this.outTimeline} />
        <ScrollRef
          name="Stack"
          ref={this.scroller}
          scrollEnabled={false}
          horizontal={false}
          focusable={false}
        >
          <View>{Config.isRoku ? [lists[currentListIndex], nullList] : lists}</View>
        </ScrollRef>

        <ViewRef name="Nav">
          <Timeline name="In" ref={this.navInTimeline} />
          <Timeline name="Out" />
        </ViewRef>

        <ViewRef name="Nav-Logo">
          <Timeline name="Loop" loop={true} />
        </ViewRef>
      </Composition>
    );
  }
}

const mapStateToProps = (store: AurynAppState) => ({
  discover: store.tmdbReducer.discover.data,
  movies: store.tmdbReducer.movies.data,
  tv: store.tmdbReducer.tv.data,
});

export default withNavigationFocus(connect(mapStateToProps)(Lander));
export { Lander as LanderTest };
