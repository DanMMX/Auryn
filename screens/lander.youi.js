import React, { Component } from 'react';
import { Composition, ViewRef, ScrollRef, ButtonRef, View, FocusManager, BackHandler } from '@youi/react-native-youi';
import { Timeline, ToggleGroup, List } from '../components';
import { withNavigationFocus, NavigationActions } from 'react-navigation';
import { connect } from 'react-redux';
import { tmdb, cache } from '../actions';
import PropTypes from 'prop-types';

@connect(store => ({
  discover: store.tmdbReducer.discover.data,
  movies: store.tmdbReducer.movies.data,
  tv: store.tmdbReducer.tv.data,
}))
class Lander extends Component {
  constructor(props) {
    super(props);
    this.state = {
      currentListIndex: 0,
    };
    this.lists = [];
    this.lastFocusItem = null;
    this.navButtonNames = ['Discover', 'Movies', 'Shows', 'Live'];
  }

  componentDidMount() {
    this.focusListener = this.props.navigation.addListener('didFocus', () => {

      this.backHandlerListener = BackHandler.addEventListener('hardwareBackPress', () => true);

      if (this.lastFocusItem) {
        FocusManager.enableFocus(this.lastFocusItem);
        FocusManager.focus(this.lastFocusItem);
      }

      // CES
      if (this.adPressed) return;
      // END CES

      if (this.landerInTimeline && this.navInTimeline) {
        this.landerInTimeline.play();
        this.navInTimeline.play(1);
      }


    });
    this.blurListener = this.props.navigation.addListener('didBlur', () => this.backHandlerListener.remove());
  }

  componentWillUnmount() {
    this.focusListener.remove();
    this.blurListener.remove();
  }

  navigateToScreen = screen => {
    console.log('NAVIGATE', `From Lander to ${screen}`);
    const navigateAction = NavigationActions.navigate({
      routeName: screen,
    });

    this.outTimeline.play().then(() => this.props.navigation.dispatch(navigateAction));
  }

  scrollToViewByIndex = (index, animated = true) => {
    for (let i = 0; i < this.lists.length; i++)
      FocusManager.setNextFocus(this.menuButtons.getButtonRef(i), this.lists[index], 'down');
    FocusManager.setNextFocus(this.searchButton, this.lists[index], 'down');
    FocusManager.setNextFocus(this.profileButton, this.lists[index], 'down');
    FocusManager.setNextFocus(this.lists[index], this.menuButtons.getButtonRef(index), 'up');

    this.setState({ currentListIndex: index });
    this.scroller.scrollTo({
      x: 0,
      y: index * 900,
      animated,
    });
  }

  onFocusItem = (ref, id, type) => {
    this.props.dispatch(cache.saveDetailsByIdAndType(id, type));
    this.lastFocusItem = ref;

    if (ref.props.shouldChangeFocus === false) return;

    FocusManager.setNextFocus(ref, this.menuButtons.getButtonRef(this.state.currentListIndex), 'up');
    for (let index = 0; index < this.lists.length; index++)
      FocusManager.setNextFocus(this.menuButtons.getButtonRef(index), ref, 'down');

    FocusManager.setNextFocus(this.searchButton, ref, 'down');
    FocusManager.setNextFocus(this.profileButton, ref, 'down');
  }

  onPressItem = (id, type, ref) => {
    // CES
    this.adPressed = false;
    if (type === 'Ad') {
      this.adPressed = true;
      this.props.navigation.dispatch(NavigationActions.navigate({ routeName: 'AdOverlay' }));
      return;
    }
    // END CES
    this.lastFocusItem = ref;
    console.log(id);
    const navigateAction = NavigationActions.navigate({
      routeName: 'PDP',
      params: {
        id,
        type,
      },
    });
    this.props.dispatch(tmdb.getDetailsByIdAndType(id, type));
    this.outTimeline.play().then(() => this.props.navigation.dispatch(navigateAction));
  }

  render() { // eslint-disable-line max-lines-per-function, max-statements
    return (
      <Composition source="Auryn_Lander">
        <ToggleGroup
          focusable={this.props.isFocused}
          prefix="Btn-Nav-"
          names={this.navButtonNames}
          onPressItem={this.scrollToViewByIndex}
          ref={ref => this.menuButtons = ref}
        />
        <ButtonRef
          name="Btn-Nav-Search"
          focusable={this.props.isFocused}
          ref={ref => this.searchButton = ref}
          onPress={() => this.navigateToScreen('Search')}
        />
        <ButtonRef
          name="Btn-Nav-Profile"
          focusable={this.props.isFocused}
          ref={ref => this.profileButton = ref}
          onPress={() => this.navigateToScreen('Profile')}
        />
        <Timeline name="LanderIn"
          onLoad={timeline => {
            this.landerInTimeline = timeline;
            this.landerInTimeline.play();
          }}
        />
        <Timeline name="LanderOut" ref={timeline => this.outTimeline = timeline} />
        <ScrollRef
          name="Stack"
          ref={scroller => this.scroller = scroller}
          scrollEnabled={false}
          horizontal={false}
          focusable={false}
        >
          <View>
            <Composition source="Auryn_Container-Discover">
              <List
                name="Discover"
                type="Discover"
                data={this.props.discover}
                ref={ref => this.lists[0] = ref}
                focusable={this.props.isFocused && this.state.currentListIndex === 0}
                onFocusItem={this.onFocusItem}
                onPressItem={this.onPressItem}
              />
            </Composition>
            <Composition source="Auryn_Container-Movies">
              <List
                name="Movies"
                type="Movies"
                data={this.props.movies}
                ref={ref => this.lists[1] = ref}
                focusable={this.props.isFocused && this.state.currentListIndex === 1}
                onFocusItem={this.onFocusItem}
                onPressItem={this.onPressItem}
              />
            </Composition>
            <Composition source="Auryn_Container-Shows">
              <List
                name="Shows"
                type="Shows"
                data={this.props.tv}
                ref={ref => this.lists[2] = ref}
                focusable={this.props.isFocused && this.state.currentListIndex === 2}
                onFocusItem={this.onFocusItem}
                onPressItem={this.onPressItem}
              />
            </Composition>
            <Composition source="Auryn_Container-Live">
              <List
                name="Live"
                type="Live"
                data={this.props.tv.slice(0, 2)}
                ref={ref => this.lists[3] = ref}
                focusable={this.props.isFocused && this.state.currentListIndex === 3}
                onFocusItem={this.onFocusItem}
                onPressItem={this.onPressItem}
              />
            </Composition>
          </View>
        </ScrollRef>

        <ViewRef name="Nav">
          <Timeline name="In" onLoad={timeline => {
            this.navInTimeline = timeline;
            this.navInTimeline.play();
            FocusManager.focus(this.menuButtons.getButtonRef(0));
          }}
          />
          <Timeline name="Out" ref={timeline => this.navOutTimeline = timeline} />
        </ViewRef>

        <ViewRef name="Nav-Logo">
          <Timeline name="Loop" loop={true} />
        </ViewRef>

      </Composition>
    );
  }
}

export default withNavigationFocus(Lander);

Lander.propTypes = {
  navigation: PropTypes.object,
  dispatch: PropTypes.func,
  isFocused: PropTypes.bool,
  discover: PropTypes.array.isRequired,
  movies: PropTypes.array.isRequired,
  tv: PropTypes.array.isRequired,
};
