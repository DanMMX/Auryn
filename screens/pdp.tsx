/**
 * Copyright (c) You i Labs Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import * as React from 'react';
import { withNavigationFocus, NavigationActions, NavigationEventSubscription, NavigationFocusInjectedProps } from 'react-navigation';
import {
  BackHandler,
  ButtonRef,
  Composition,
  ImageRef,
  TextRef,
  ViewRef,
  FocusManager,
} from '@youi/react-native-youi';
import { View, NativeEventSubscription } from 'react-native';
import { connect } from 'react-redux';
import { Timeline, List, BackButton } from '../components';
import { Asset, AssetType } from '../adapters/asset';
import { Config } from '../config';
import { AurynAppState } from '../reducers';
import { ListType } from '../components/list';
import { prefetchDetails, getDetailsByIdAndType } from '../actions/tmdbActions';
import { getVideoSourceByYoutubeId } from '../actions/youtubeActions';
import { ListItemFocusEvent, ListItemPressEvent } from '../components/listitem';

type PdpDispatchProps = typeof mapDispatchToProps;

interface PdpProps extends NavigationFocusInjectedProps, PdpDispatchProps {
  asset: Asset;
  fetched: boolean;
}

class PDP extends React.Component<PdpProps> {
  outTimeline = React.createRef<Timeline>();

  videoOutTimeline = React.createRef<Timeline>();

  videoInTimeline = React.createRef<Timeline>();

  contentOutTimeline = React.createRef<Timeline>();

  contentInTimeline = React.createRef<Timeline>();

  posterButton = React.createRef<ButtonRef>();

  focusListener!: NavigationEventSubscription;

  blurListener!: NavigationEventSubscription;

  backHandlerListener!: NativeEventSubscription;

  navigateBack = async () => {
    if (this.outTimeline.current)
      await this.outTimeline.current.play();

    if (Config.isRoku)
      this.props.navigation.navigate({ routeName: 'Lander' });
    else
      this.props.navigation.popToTop();

    return true;
  }

  onPressItem: ListItemPressEvent = async (id: any, type: AssetType) => {
    this.props.getDetailsByIdAndType(id, type);
    if (this.contentOutTimeline.current) await this.contentOutTimeline.current.play();
    await this.props.navigation.navigate({ routeName: 'PDP', params: { id, type }, key: id });
    if (this.posterButton.current) FocusManager.focus(this.posterButton.current);
    if (this.contentInTimeline.current) this.contentInTimeline.current.play();
  }

  onFocusItem: ListItemFocusEvent = (id: any, type: AssetType) => {
    this.props.prefetchDetails(id, type);
  }

  componentDidMount() {

    this.focusListener = this.props.navigation.addListener('didFocus', () => {
      this.backHandlerListener = BackHandler.addEventListener('hardwareBackPress', this.navigateBack);

      if (this.videoOutTimeline.current) this.videoOutTimeline.current.play();
    });

    this.blurListener = this.props.navigation.addListener('didBlur', () => this.backHandlerListener.remove());
  }

  componentWillUnmount() {
    this.focusListener.remove();
    this.blurListener.remove();
    this.backHandlerListener.remove();
  }

  shouldComponentUpdate(nextProps: PdpProps) {
    // Re-render if lost/gained focus
    if (nextProps.isFocused !== this.props.isFocused)
      return true;

    if (nextProps.fetched && !this.props.fetched)
      return true;

    // Only render if the asset.id matches the requested pdp asset id
    return nextProps.asset.id === nextProps.navigation.getParam('id');
  }

  playVideo = async () => {
    if (this.videoInTimeline.current)
      await this.videoInTimeline.current.play();
    this.props.navigation.dispatch(NavigationActions.navigate({
      routeName: 'Video',
    }));
  }

  onFocusPoster = () => {
    this.props.getVideoSourceByYoutubeId(this.props.asset.youtubeId);
  }

  onLoad = () => {
    if (this.posterButton.current)
      FocusManager.focus(this.posterButton.current);
  }

  render() { // eslint-disable-line max-lines-per-function
    const { asset, fetched, isFocused } = this.props;

    if (!fetched || !isFocused)
      return <View/>;

    return (
      <Composition source="Auryn_PDP">
        <Timeline name="VideoIn" ref={this.videoInTimeline} />
        <Timeline name="VideoOut" ref={this.videoOutTimeline} />
        <Timeline name="PDPIn" playOnLoad />
        <Timeline name="PDPOut" ref={this.outTimeline} />

        <ViewRef name="PDP-Scroller" visible={isFocused && fetched}>
          <BackButton
            focusable={isFocused}
            onPress={this.navigateBack}
          />
          <List
            name="List-PDP"
            type={ListType.SmallBackdrop}
            data={asset.similar}
            focusable={isFocused}
            onPressItem={this.onPressItem}
            onFocusItem={this.onFocusItem}
          />

          <Timeline
            name="ContentIn"
            ref={this.contentInTimeline}
            onLoad={ref => ref.play()} />
          <Timeline name="ContentOut" ref={this.contentOutTimeline} />

          <ButtonRef
            name="Btn-Poster-Large"
            focusable={isFocused}
            onPress={this.playVideo}
            ref={this.posterButton}
            onLoad={this.onLoad}
            onFocus={this.onFocusPoster}
          >
            <ImageRef
              name="Image-Dynamic"
              source={{ uri: asset.thumbs.Poster }}
            />
          </ButtonRef>

          <ImageRef
            name="Image-Dynamic-Background"
            source={{ uri: asset.images.Backdrop }}
          />

          <ViewRef name="Layout-PDP-Meta">
            <TextRef name="Text-Title" text={asset.title} />
            <TextRef name="Text-Overview" text={asset.details} />
            <TextRef name="Text-Featured" text={asset.extra} />
            <Timeline name="In2" playOnLoad />
          </ViewRef>

        </ViewRef>
      </Composition>
    );
  }
}

const mapStateToProps = (store: AurynAppState) => ({
  asset: store.tmdbReducer.details.data,
  fetched: store.tmdbReducer.details.fetched,
});

const mapDispatchToProps = {
  getVideoSourceByYoutubeId,
  prefetchDetails,
  getDetailsByIdAndType,
};

export default withNavigationFocus(connect(mapStateToProps, mapDispatchToProps)(PDP as any));
export { PDP as PdpTest };
