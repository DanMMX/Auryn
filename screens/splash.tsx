/**
 * Copyright (c) You i Labs Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import * as React from 'react';
import { Composition, ViewRef } from '@youi/react-native-youi';
import { View, StyleSheet } from 'react-native';
import { Timeline } from '../components';
import { tmdb } from '../actions';
import { NavigationActions, NavigationScreenProps } from 'react-navigation';
import { connect, DispatchProp } from 'react-redux';
import { Config } from '../config';
import { AnyAction } from 'redux';

interface SplashProps extends NavigationScreenProps, DispatchProp<AnyAction> {
  fetched: boolean;
};

class Splash extends React.Component<SplashProps> {
  outTimeline: React.RefObject<Timeline> = React.createRef<Timeline>();

  componentDidMount() {
    this.props.dispatch(tmdb.getDiscover());
    this.props.dispatch(tmdb.getMovies());
    this.props.dispatch(tmdb.getTv());
  }

  async componentDidUpdate() {
    if (this.props.fetched && !Config.isRoku) {
      if (!Config.isRoku && this.outTimeline.current)
        await this.outTimeline.current.play();
      const landerNavigationAction = NavigationActions.navigate({
        routeName: 'Lander',
      });
      this.props.navigation.dispatch(landerNavigationAction);
    }
  }

  render() {
    return (
      <View style={styles.container}>
        <Composition source="Auryn_Splash">
          <Timeline name="SplashIn" playOnLoad />
          <Timeline name="SplashOut" ref={this.outTimeline} />
          <ViewRef name="Loader">
            <Timeline name="Loop" playOnLoad={!Config.isRoku} />
          </ViewRef>
        </Composition>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#143672',
  },
});

const mapStateToProps = store => ({
  fetched:
    store.tmdbReducer.discover.fetched
    && store.tmdbReducer.movies.fetched
    && store.tmdbReducer.tv.fetched,
});

export default connect(mapStateToProps)(Splash);
export { Splash as SplashTest };
