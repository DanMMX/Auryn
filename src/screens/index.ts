/**
 * Copyright (c) You i Labs Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

export { Splash }   from './splash';
export { Lander }   from './lander';
export { Pdp }      from './pdp';
export { Search }   from './search';
export { Profile }  from './profile';
export { VideoScreen as Video }    from './video';
