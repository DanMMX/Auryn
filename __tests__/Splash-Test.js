/**
 * Copyright (c) You i Labs Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import React from 'react';
import { SplashTest } from '../screens/splash';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(<SplashTest navigation={global.navigation} dispatch={jest.fn()}/>).toJSON();
  expect(tree).toMatchSnapshot();
});
