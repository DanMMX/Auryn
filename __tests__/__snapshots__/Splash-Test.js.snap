// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#143672",
      "flex": 1,
    }
  }
>
  <Composition
    onTimelinesLoaded={[Function]}
    source="Auryn_Splash"
  >
    <TimelineRef
      direction="forward"
      loop={false}
      name="SplashIn"
      onAborted={[Function]}
      onCompleted={[Function]}
      onCompletedForward={[Function]}
      onCompletedReverse={[Function]}
      onLoad={[Function]}
      onPaused={[Function]}
      onPausedForward={[Function]}
      onPausedReverse={[Function]}
      onPlay={[Function]}
      onPlayForward={[Function]}
      onPlayReverse={[Function]}
      onStarted={[Function]}
      onStartedForward={[Function]}
      onStartedReverse={[Function]}
      playOnLoad={true}
    />
    <TimelineRef
      direction="forward"
      loop={false}
      name="SplashOut"
      onAborted={[Function]}
      onCompleted={[Function]}
      onCompletedForward={[Function]}
      onCompletedReverse={[Function]}
      onLoad={[Function]}
      onPaused={[Function]}
      onPausedForward={[Function]}
      onPausedReverse={[Function]}
      onPlay={[Function]}
      onPlayForward={[Function]}
      onPlayReverse={[Function]}
      onStarted={[Function]}
      onStartedForward={[Function]}
      onStartedReverse={[Function]}
      playOnLoad={false}
    />
    <ViewRef
      name="Loader"
      onLoad={[Function]}
      onTimelinesLoaded={[Function]}
    >
      <TimelineRef
        direction="forward"
        loop={true}
        name="Loop"
        onAborted={[Function]}
        onCompleted={[Function]}
        onCompletedForward={[Function]}
        onCompletedReverse={[Function]}
        onLoad={[Function]}
        onPaused={[Function]}
        onPausedForward={[Function]}
        onPausedReverse={[Function]}
        onPlay={[Function]}
        onPlayForward={[Function]}
        onPlayReverse={[Function]}
        onStarted={[Function]}
        onStartedForward={[Function]}
        onStartedReverse={[Function]}
        playOnLoad={true}
      />
    </ViewRef>
  </Composition>
</View>
`;
